core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/DiagnosticLog.java

```diff
@@ -346,20 +306,12 @@ public DiagnosticLog build() {
                 // required. There shouldn't be a diagnostic log without a result message and input.
                 throw new IllegalStateException("Either resultMessage or input must be provided.");
             }
-            if (this.logId == null) {
-                logId = UUID.randomUUID().toString();
-            }
-            if (this.recordedAt == null) {
-                recordedAt = Instant.now();
-            }
-            if (this.requestId == null) {
-                requestId = MDC.get(CORRELATION_ID_MDC);
-            }
-            if (this.flowId == null) {
-                flowId = MDC.get(FLOW_ID_MDC);
-            }
-            if (this.logLevel == null) {
-                logLevel = LogLevel.BASIC;
+            logId = UUID.randomUUID().toString();
+            recordedAt = Instant.now();
+            requestId = MDC.get(CORRELATION_ID_MDC);
+            flowId = MDC.get(FLOW_ID_MDC);
```

Comment:
In the previous build methods, we first check whether a value for these two exists, and set the MDC values only if they do not exist. Is this change of always reading it from MDC an expected one?

Reply:
Previously, the only place that these two were set is by the framework (central logger component), and it's always using the values in the MDC. As I mentioned earlier, the null checks have been introduced in the previous kernel release and I thought we should allow to set any custom value to these two attributes. Then in a recent discussion, we agreed that we should always use the values in MDC for these two attributes.